src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -45,107 +58,67 @@ export const DropdownMenu: React.FC<Props> = ({
   children,
   className,
   isActive,
-  handleOnKeyDown,
+  handleOnEscDown,
   ...other
 }) => {
-  const childRefs = useRef<Array<HTMLLIElement>>([]);
+  const refs = useRef<Refs>({
+    set: new Set(),
+    list: [],
+  });
+
+  let focusIndex = 0;
+  const focusItem = (index: number) => {
+    refs.current.list[index]
+      ?.querySelector<HTMLButtonElement | HTMLAnchorElement>(':first-child')
+      ?.focus();
+  };
 
   useEffect(() => {
-    setTimeout(() => {
-      if (isActive && childRefs) {
-        childRefs.current[0]
-          ?.querySelector<HTMLButtonElement | HTMLAnchorElement>(':first-child')
-          ?.focus();
-      }
-    }, 1);
+    if (isActive && refs.current.list.length) {
+      focusItem(0);
+    }
   }, [isActive]);
 
-  /**
-   * On KeyDown
-   * 1) Find active item. If there isn't one, do nothing on Keydown
-   * 2) Set active item, next item, and previous item.
-   * 3) If right or down arrow key keyed, focus on next item.
-   * 4) If left or up arrow key keyed, focus on the previous item.
-   */
   const onKeyDown = (e: KeyboardEvent<HTMLUListElement>) => {
     if ([ESCAPE_KEYCODE].includes(e.key)) {
```

Comment:
Just curious – do you understand why here they're using `includes` instead of `==`? Maybe they copy/pasted this from somewhere else that was checking against multiple keycodes..

Reply:
Also could we flatten these if statements into one? Ie
```
if (e.key === ESCAPE_KEYCODE && handleOnEscDown) {
  handleOnEscDown(e);
}
```

Reply:
Good point, I'll make the conditionals more readable